istioctl/pkg/tag/generate.go

```diff
@@ -130,6 +131,13 @@ func Generate(ctx context.Context, client kube.ExtendedClient, opts *GenerateOpt
 		// TODO(Monkeyanator) should extract the validationURL from revision's validating webhook here. However,
 		// to ease complexity when pointing default to revision without per-revision validating webhook,
 		// instead grab the endpoint information from the mutating webhook. This is not strictly correct.
+		if tagWhConfig.URL != "" {
+			webhookURL, err := url.Parse(tagWhConfig.URL)
+			if err == nil {
+				webhookURL.Path = "/validate"
+				tagWhConfig.URL = webhookURL.String()
```

Comment:
@Monkeyanator This fix is still "not strictly correct" either, but the original code is incorrect. At some point I think this should really be fixed properly.

Reply:
feel that the fix will make the logic even more complex. don't we have some kind of very clear logic to indicate if the webhook should be touched? such as check if istiod is included or this is a remote cluster install or some other conditions? I have tried this case, install remote cluster which has webhook installed and the external istiod also injected the caBundles to the webhook, then when I install a gateway, it completely messed up the webhook, it made the external istiod think there is a leader election with different root ca and produced unauthorized requests. After awhile digging, I figured it was due to the install messing with the webhooks. was not easy to figure out why.

Reply:
@litong this change has nothing to do with what you're describing. It's simply fixing the incorrect URL in the default validating webhook configuration. Line 207 in the other file is trying to prevent generating the webhook when all that's being installed is a gateway.

Reply:
@frankbu ok.